lark/tree_templates.py

```diff
@@ -141,16 +144,14 @@ def translate(t1: Template, t2: Template, tree: TreeOrCode):
     return tree
 
 
-
 class TemplateTranslator:
     """Utility class for translating a collection of patterns
     """
 
-    def __init__(self, translations: Mapping[TreeOrCode, TreeOrCode]):
-        assert all( isinstance(k, Template) and isinstance(v, Template) for k, v in translations.items() )
+    def __init__(self, translations: Mapping[Template, Template]):

```

Comment:
Since the assert statement checks that the dictionary actually contains `Template`, I considered that the source of truth and corrected the annotations to match.

Reply:
Yeah, you're right.

But why remove the assert?

Reply:
I guess my thought process was: if the type annotations are correct, type checkers will catch cases where the wrong type is used. However, that assumes everyone is using type checking, which is not accurate. I'll add it back